### PR TITLE
Feature/choose notes search

### PR DESCRIPTION
**Pull request by JohnPadgettCivica**
For,
https://dev.azure.com/civica-digital/EA%20-%20WEEE/_workitems/edit/51002
https://dev.azure.com/civica-digital/EA%20-%20WEEE/_workitems/edit/51003

### CODE REVIEW

File: src/EA.Weee.DataAccess/DataAccess/EvidenceDataAccess.cs [hunk 1]
```diff
@@ -178,9 +179,12 @@ public async Task<int> GetComplianceYearByNotes(List<Guid> evidenceNoteIds)
                                                          n.Status.Value == NoteStatus.Approved.Value &&
                                                          n.ComplianceYear == complianceYear);
 
-            if (searchRef.HasValue)
+            if (!string.IsNullOrWhiteSpace(searchRef))
             {
-                filteredNotes = filteredNotes.Where(n => n.Reference == searchRef.Value);
+                var regex = new Regex("^[E?|e?|][1-9]+");
```
Comment (meglenapnovoselska) [inline src/EA.Weee.DataAccess/DataAccess/EvidenceDataAccess.cs:184]: I think it matches if searchRef = e3tr4fd3d, is it okay to have letters in the ref?

File: src/EA.Weee.Web/Areas/Scheme/Views/Shared/_TransferFromSummaryPartial.cshtml [hunk 2]
```diff
@@ -105,6 +105,48 @@
...
+                }
+                else
+                {
+                    clearFilterRouteValues = new { pcsId = Model.PcsId, complianceYear = Model.ComplianceYear, page = 1 };
+                }
+
+            }
+            <div class="govuk-grid-column-one-third govuk-!-padding-top-9">
+                @Html.ActionLink("Clear filter", postAction, controller,
+                    clearFilterRouteValues, new { @class = "link-submit govuk-link govuk-link--no-visited-state", @style = "float: right;" })
```
[the remark this answers is not in the document]
Comment (JohnPadgettCivica) [reply src/EA.Weee.Web/Areas/Scheme/Views/Shared/_TransferFromSummaryPartial.cshtml]: done...

File: src/EA.Weee.Web.Tests.Unit/Areas/Scheme/Controllers/OutgoingTransfersControllerTests.cs [hunk 3]
```diff
@@ -3457,8 +3452,10 @@ public async Task SelectEvidenceNotePost_GivenInvalidModel_ShouldRetrieveTransfe
                 A<GetTransferEvidenceNoteForSchemeRequest>.That.Matches(r => r.EvidenceNoteId == model.EditEvidenceNoteId))).MustHaveHappenedOnceExactly();
         }
 
-        [Fact]
-        public async Task SelectEvidenceNotePost_GivenInvalidModel_AvailableTransferNotesShouldBeRetrieved()
+        [Theory]
+        [InlineData(null)]
```
[the remark this answers is not in the document]
Comment (JohnPadgettCivica) [reply src/EA.Weee.Web.Tests.Unit/Areas/Scheme/Controllers/OutgoingTransfersControllerTests.cs:3460]: done...

File: src/EA.Weee.DataAccess/DataAccess/EvidenceDataAccess.cs (hunk 1 above)
Comment (JohnPadgettCivica) [reply src/EA.Weee.DataAccess/DataAccess/EvidenceDataAccess.cs]: Updated the regex

**Pull request merged**